README.md

```diff
@@ -55,3 +55,77 @@ See `k8s-jenkins-cac.tf` in your project.
+
+---
+
+#### Environment Config
+
+An `environment` is the set of machines where your microservices will run. For example: `development`, `staging` and `production` (or `de-prod`, `jp-prod`, and `pt-prod` which would all be `production` environments - and belong to the `production` environment group - but are machines in different countries).
+
+In a cloud provider like GCP, we recommend separating the environments on a project level. This means that one GCP project correlates to one kuberpult environment 1:1 - although this is not a technical requirement.
+
+Environments are also called `stages` - but in kuberpult we stick to `environments`, short `envs`.
+
+The config for an environment is stored in a json file called `config.json`. This file would go into an environment in the manifests repository like this: `environments/development/config.json` (in this example the `config.json` file would dictate the configuration for the `development` environment)
+
+In the `config.json` file there are 3 main fields:
+- [Upstream](#upstream)  `"upstream"`
+- [ArgoCD](#argocd)    `"argocd"`
+- [EnvironmentGroup](#environment-group) `"environmentGroup"`
+
+##### Upstream:
+
+The `"upstream"` field can have one of the two options (cannot have both):
+  - `latest`: can only be set to `true` which means that any kuberpult will deploy the latest version of an application to this environment
+  - `environment`: has a string which is the name of another environment. Following the chain of upstream environments would take you to the one with `"latest": true`. This is used in release trains: when a release train is run in an environment, it will pull the version from the environment's upstream environment.
+
+##### ArgoCd: 
+
+The `"argocd"` field has a few subfields:
+- `"accessList"`:  
+  Is a list of objects that describe which resources are allowed in the cluster ([ArgoCD Docs](https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#projects)).
+  Note that this is only necessary when argoCd cannot apply a certain resource on the cluster. Most common resources (Pods, Configmaps, Cronjobs, Deployments...) just work out of the box.
+  It has the following fields:
+  - `"group"`: relates to the Kubernetes API group without the version
+  - `"kind"`: the resource type in kubernetes
+
+- `"destination"` (**Mandatory**):
+  > Defines which Kubernetes cluster/namespace to deploy to
+  >
+  > \- [Template ArgoCD Docs](https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Template/#template-fields)
+
+  It has the following fields:
+  - `"name"` (**Mandatory**): Name of the cluster (within Argo CD) to deploy to
+  - `"server"` (**Mandatory**): API Server URL for the cluster (Example: `"https://kubernetes.default.svc"`)
+  - `"namespace"`: Target namespace in which to deploy the manifests from source (Example: `"my-app-namespace"`)
```

Comment:
there are more fields in the destination https://github.com/freiheit-com/kuberpult/blob/3619b6bf89eb4a070449213ac93e1d9433c0c9cd/services/cd-service/pkg/config/config.go#L47

Reply:
Good catch! Thank you!